### ISSUE TITLE
Multiple validation functions in rule doesn't work

### ISSUE THREAD

**Issue by 1yuv** [medic/cht-core#8806]
**Describe the bug**
If you apply Logical operation to multiple [cht validation functions](https://docs.communityhealthtoolkit.org/apps/reference/app-settings/#cht-validation-functions) it doesn't work. 

**To Reproduce**
1. Prepare an app settings with forms for validations. 
2. On config, ensure that two validations are combined together, for example we are trying to check whether L or G form has been submitted previously before f1 form is submitted. Add following validation property in `f1` form. 

```js
validations = [
      {
        property: 'parent_id',
        rule: '"exists('L','patient_id') || exists('G','patient_id')',
        message: [
          {
            content: 'Pregnancy not registered already',
            locale: 'en',
          },
        ],
      }
```
3. Compile app setting, upload to instance. 
4. Ensure patient doesn't have `G` or `L` form submitted already.
5. Submit `f1` form. Since `L` or `G` form doesn't exist, it will work properly and report `Pregnancy not registered already`. This is expected. 
6. Now, submit `G` form for this patient. 
7. After submitting G form, submit `f1` form again. 
8. This will again throw error `pregnancy not registered already`.

**Expected behavior**
1. In step 8 above, it should not throw an error since `G` form is already submitted. 

**Environment**
- Instance: Any
- Browser: Any
- Client platform: Any
- App: Sentinel
- Version: 4.5.0, but this likely exists in all previous supported versions. 

**Additional context**
If this issue is fixed, we won't need to work on #8402 .

**Comment by dianabarsan** [comment]
I'm afraid this is known and not a bug: https://github.com/medic/cht-core/blob/master/shared-libs/validation/src/validation.js#L230-L231

> Validation settings may consist of Pupil.js rules and custom rules.
> These cannot be combined as part of the same rule.

I think the best course of action here is to: 
1. clearly document this limitation
2. consider adding the possibility of combining rules as a feature.

**Comment by 1yuv** [comment]
Thanks @dianabarsan , the [documentation](https://docs.communityhealthtoolkit.org/apps/reference/app-settings/#rules) did not mention that custom rules can not be combined, so I expected it to be desired feature. 

>Validation settings may consist of Pupil.js rules and rules specific to the CHT. These two types of rules cannot be combined as part of the same rule.

It'd be good to have multiple custom rules to be combined together which would allow more flexibility to validation.

**Comment by garethbowen** [comment]
This is scheduled to be fixed in 4.15.0. For configurations to work with older versions of the CHT, please note that
1. you cannot use operators to combine two "Medic" rules, and
2. you cannot use operators to combine a "Medic" rule and a "Pupil" rule

The workaround is to specify two separate validation rules for the same property.

For posterity, the Pupil rules are:

- `equals`
- `iEquals`
- `sEquals`
- `siEquals`
- `lenMin`
- `lenMax`
- `lenEquals`
- `min`
- `max`
- `between`
- `in`
- `required`
- `optional`
- `numeric`
- `integer`
- `alpha`
- `alphaNumeric`
- `email`
- `regex`
- `equalsTo`

And the Medic rules are:

- `unique`
- `uniquePhone`
- `validPhone`
- `uniqueWithin`
- `exists`
- `isISOWeek`
- `isAfter`
- `isBefore`
